packages/data-stores/src/plans/resolvers.ts

```diff
@@ -41,90 +47,91 @@ function getMonthlyPrice( plan: APIPlan ) {
-	const prices: Record< string, string > = WPCOMPlans.reduce( ( acc, plan ) => {
-		acc[ plan.product_slug ] = getMonthlyPrice( plan );
-		return acc;
-	}, {} as Record< string, string > );
-
-	yield setPrices( prices );
-}
-
-const mapShortNameToProductSlug: Record< string, string > = {
-	Free: PLAN_FREE,
-	Personal: PLAN_PERSONAL,
-	Premium: PLAN_PREMIUM,
-	Business: PLAN_BUSINESS,
-	eCommerce: PLAN_ECOMMERCE,
-};
-
-export function* getPlansDetails( locale = 'en' ) {
-	try {
-		const { body: rawPlansDetails } = yield fetchAndParse(
-			`https://public-api.wordpress.com/wpcom/v2/plans/details?locale=${ encodeURIComponent(
-				locale
-			) }`,
-			{
-				mode: 'cors',
-				credentials: 'omit',
-			}
-		);
-
-		const plans: Record< string, Plan > = {};
-		const features: Record< string, PlanFeature > = {};
-
-		rawPlansDetails.features.forEach( ( feature: Record< string, string > ) => {
-			features[ feature.id ] = {
-				id: feature.id,
-				name: feature.name,
-				description: feature.description,
-				type: feature.type ?? 'checkbox',
-			};
-		} );
-
-		rawPlansDetails.plans.forEach( ( rawPlan: APIPlanDetail ) => {
-			const plan: Plan = {
-				title: rawPlan.short_name,
-				description: rawPlan.tagline,
-				productId: rawPlan.products[ 0 ].plan_id,
-				storeSlug: mapShortNameToProductSlug[ rawPlan.nonlocalized_short_name ],
-				features: rawPlan.highlighted_features,
-				pathSlug: rawPlan.nonlocalized_short_name?.toLowerCase(),
-				featuresSlugs: rawPlan.features.reduce(
-					( acc: Record< string, boolean >, cur: string ) => ( { ...acc, [ cur ]: true } ),
-					{}
-				),
-				storage: rawPlan.storage,
-			};
-
-			if ( 'Free' === rawPlan.nonlocalized_short_name ) {
-				plan.isFree = true;
-			}
-
-			if ( 'Premium' === rawPlan.nonlocalized_short_name ) {
-				plan.isPopular = true;
-			}
-
-			plans[ mapShortNameToProductSlug[ rawPlan.nonlocalized_short_name ] ] = plan;
-		} );
-
-		yield setPlans( plans );
-		yield setFeatures( features );
-		yield setFeaturesByType( rawPlansDetails.features_by_type );
-	} catch ( err ) {
-		// TODO: Add error handling
-		return;
+	const { body: plansFeatures } = ( yield fetchAndParse(
+		`https://public-api.wordpress.com/wpcom/v2/plans/details?locale=${ encodeURIComponent(
+			locale
+		) }`,
+		{
+			mode: 'cors',
+			credentials: 'omit',
+		}
+	) ) as { body: DetailsAPIResponse };
+
+	const plans = plansProductSlugs.reduce( ( plans, slug ) => {
+		const pricedPlan = planPrices.find(
+			( pricedPlan ) => pricedPlan.product_slug === slug
+		) as PricedAPIPlan;
+
+		if ( ! pricedPlan ) {
+			return plans;
+		}
+
+		// each details objects has a products[] array containing all the plans' ID it covers
+		// Here we find the right details object using this product id array
+		const {
+			features: planFeaturesSlugs,
+			...planDetails
+		} = plansFeatures.plans.find( ( planDetails ) =>
+			planDetails.products.find( ( product ) => product.plan_id === pricedPlan.product_id )
```

Comment:
We no longer use short names to match details and plans! We use `product_id`. I didn't like the old mapping one bit.

But this needs D55310-code.